python/ray/rllib/models/preprocessors.py

```diff
@@ -8,7 +8,8 @@
 class Preprocessor(object):
     """Defines an abstract observation preprocessor function."""
 
-    def __init__(self, options):
+    def __init__(self, obs_space, options):
+        self.obs_space = obs_space
```

Comment:
One thing I don't quite understand is that since the preprocessor is defined on an observation space, does it essentially know what environment it is responsible for? 

If so, then what's the `transform_shape(obs_shape)` function responsible for? Why not just `transformed_shape()`? Can the observation be a completely different shape than the `obs_space` originally defined?

Reply:
Yea I was debating whether to remove transform_shape(), but we can do it in another pr.

Reply:
Ok I changed it to just define `.shape`, it is much cleaner now.